Updater/Server.cs

```diff
@@ -256,76 +269,113 @@ private static void MetadataHandler(DataPacket dataPacket, ICommunicator communi
             DirectoryMetadataComparer comparerInstance = new DirectoryMetadataComparer(metadataServer, metadataClient);
             List<MetadataDifference> differences = comparerInstance.Differences;
 
-            // Serialize and save differences to C:\temp\ folder
-            string serializedDifferences = Utils.SerializeObject(differences);
-            string tempFilePath = @$"{Server.s_serverDirectory}\differences.xml";
-
-            if (string.IsNullOrEmpty(serializedDifferences))
+            // Check if the sync up is invalid
+            // If it is invalid, server will send an InvalidSync response packet to
+            // client along with list of filenames that needs to be changed in the client side
+            if (!comparerInstance.ValidateSync())
             {
-                Trace.WriteLine("[Updater] Serialization of differences failed or resulted in an empty string.");
-                return; // Exit if serialization fails
-            }
+                List<string> invalidFileNames = comparerInstance.InvalidSyncUpFiles;
 
-            try
-            {
-                Directory.CreateDirectory(Path.GetDirectoryName(tempFilePath)!);
-                File.WriteAllText(tempFilePath, serializedDifferences);
-                Server.UpdateUILogs($"Differences file saved to {tempFilePath}");
-                Trace.WriteLine($"[Updater] Differences file saved to {tempFilePath}");
-            }
-            catch (Exception ex)
-            {
-                Trace.WriteLine($"[Updater] Error saving differences file: {ex.Message}");
-            }
+                FileContent fileContentToSend = new FileContent(
+                        "invalidFileNames.list",
+                        Utils.SerializeObject(invalidFileNames)
+                        );
 
-            // Prepare data to send to client
-            List<FileContent> fileContentsToSend = new List<FileContent>
+                DataPacket dataPacketToSend = new DataPacket(
+                        DataPacket.PacketType.InvalidSync,
+                        new List<FileContent> { fileContentToSend }
+                        );
+
+                string serializedDataPacket = Utils.SerializeObject(dataPacketToSend);
+
+                try
                 {
-                    // Added difference file to be sent to client
-                    new FileContent("differences.xml", serializedDifferences)
-                };
+                    Server.UpdateUILogs($"Sending files to client and waiting to recieve files from client {clientId}");
+                    communicator.Send(serializedDataPacket, "Server", clientId);
```

Comment:
The string should be "FileTransferHandler", not "Server"
Only the string value matters for defining a unique handler, not the INotificationHandler, any doubts we can discuss later

Reply:
kk